lib/node/httpStack.js

```diff
@@ -5,22 +5,44 @@ import {Readable, Transform} from "stream";
 import throttle from "lodash.throttle";
 
 export default class NodeHttpStack {
+  constructor(requestOptions = {}) {
+    this._requestOptions = requestOptions;
+  }
+
   createRequest(method, url) {
-    return new Request(method, url);
+    return new Request(method, url, this._requestOptions);
   }
 
   getName() {
     return "NodeHttpStack";
+    // return this.constructor.name;
```

Comment:
Please remove this unused line.

Reply:
wish I could've used that approach, however, I'm not sure if it's a TS feature or an ES feature. We can leave a string for now.